packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol

```diff
@@ -696,4 +708,220 @@ contract BridgeFacet is BaseConnextFacet {
       }
     }
   }
+
+  /**
+   * @notice Uses Aave Portals to provide fast liquidity
+   */
+  function _executePortalTransfer(
+    bytes32 _transferId,
+    uint256 _fastTransferAmount,
+    address _local,
+    address _router
+  ) internal returns (uint256, address) {
+    // Calculate local to adopted swap output if needed
+    (uint256 userAmount, address adopted) = AssetLogic.calculateSwapFromLocalAssetIfNeeded(_local, _fastTransferAmount);
+
+    IAavePool(s.aavePool).mintUnbacked(adopted, userAmount, address(this), AAVE_REFERRAL_CODE);
+
+    // Improvement: Instead of withdrawing to address(this), withdraw directly to the user or executor to save 1 transfer
+    IAavePool(s.aavePool).withdraw(adopted, userAmount, address(this));
+
+    s.aavePortalsTransfers[_transferId] = userAmount;
+
+    emit AavePortalMintUnbacked(_transferId, _router, adopted, userAmount);
+
+    return (userAmount, adopted);
+  }
+
+  /**
+   * @notice Parses the message and process the transfer
+   * @dev Will mint the tokens if the token originates on a remote origin
+   * @return The message amount
+   * @return The message token
+   * @return The message transfer id
+   */
+  function _reconcileProcessMessage(bytes memory _message)
+    internal
+    returns (
+      uint256,
+      address,
+      bytes32,
+      bytes32
+    )
+  {
+    // parse tokenId and action from message
+    bytes29 msg_ = _message.ref(0).mustBeMessage();
+    bytes29 tokenId = msg_.tokenId();
+    bytes29 action = msg_.action();
+
+    // assert the action is valid
+    if (!action.isTransfer()) {
+      revert BridgeFacet__reconcile_invalidAction();
+    }
+
+    // load the transferId
+    bytes32 transferId = action.transferId();
+
+    // ensure the transaction has not been handled
+    if (s.reconciledTransfers[transferId]) {
```

Comment:
Ik you did not write this code @liu-zhipeng but please move this check to be the very first thing in the method. Just in case there's a race to reconcile, somehow, we want the revert on the redundant tx to cost as little as possible